tests/external/helpers.cpp

```diff
@@ -80,3 +83,41 @@ NamedTempPath::NamedTempPath(std::string extension) {
 NamedTempPath::~NamedTempPath() {
     remove(path_.c_str());
 }
+
+static bool FAIL_NEXT_ALLOCATION = false;
+
+void fail_next_allocation() {
+    FAIL_NEXT_ALLOCATION = true;
+}
+
+// Replace global operator new/operator delete so that we can force them to
```

Comment:
One can do that???

Reply:
Yay, C++! https://en.cppreference.com/w/cpp/memory/new/operator_new, "Global replacement" section

Reply:
cool